src/tests/system/tests/test_ldap.py

```diff
@@ -0,0 +1,57 @@
+""" SSSD LDAP provider tests
+
+:requirement: IDM-SSSD-REQ : LDAP Provider
+"""
+
+from __future__ import annotations
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.ldap import LDAP
+from sssd_test_framework.topology import KnownTopology
+
+
+@pytest.mark.parametrize("modify_mode", ["exop", "ldap_modify"])
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_ldap__change_password(client: Client, ldap: LDAP, modify_mode: str):
+    """
+    :title: Changing password, with "ldap_pwmodify_mode" set to "exop" and then "ldap_modify"
+    :setup:
+        1. Add user to SSSD, set his password
+        2. Allow user to change his password
+        3. Set "ldap_pwmodify_mode"
+        4. Start SSSD
+    :steps:
+        1. Check that "ldap_pwmodify_mode" is properly set
+        2. Authenticate user with old password
+        3. Change password of user to new password
+        4. Authenticate user with new password
+        5. Authenticate user with old password
+    :expectedresults:
+        1. Config is properly set
+        2. User is authenticated
+        3. Password is changed successfully
+        4. User is authenticated
+        5. User is not authenticated
+    :customerscenario: False
+    """
+    user = "user1"
+    old_pass = "Secret123"
+    new_pass = "New_password123"
+
+    ldap.user(user).add(password=old_pass)
+    ldap.aci.add('(targetattr="userpassword")(version 3.0; acl "pwp test"; allow (all) userdn="ldap:///self";)')
+    client.sssd.dom("test")["ldap_pwmodify_mode"] = modify_mode
+
+    client.sssd.start()
+
+    ldb = client.ldb.search("/var/lib/sss/db/config.ldb", "cn=test,cn=domain,cn=config")
+    assert ldb["cn=test,cn=domain,cn=config"]["ldap_pwmodify_mode"] == [modify_mode]
+
+    assert client.auth.ssh.password(user, old_pass)
+
+    with client.ssh(user, old_pass) as ssh:
+        assert client.auth.passwd(ssh).password(user, old_pass, new_pass)
```

Comment:
This looks pretty unreadable, what is going on there?
There should be at least assert message present to explain what it is asserting.

Reply:
I think that in docstring it is written pretty clear what are the steps. But I agree that an assert message would not be a bad idea.